### PR TITLE
Upgrade to Taffy 0.6

### PR DESCRIPTION
**Pull request by nicoburns**
# Objective

- Keep Taffy version up to date

Taffy 0.6 doesn't include a huge amount relevant to Bevy. But it does:

- Add the `box_sizing` style
- Expose the computed `margin` in layout
- Traitifies the `Style` struct, which opens up the possibility of using Bevy's `Style` struct directly (although Bevy currently does some style resolution at conversion time which would no longer be cached if it was used directly).
- Have a few bug fixes in the layout algorithms

## Solution

- Upgrade Taffy to `0.6.0`

## Testing

- I've run the `grid` example. All looks good.
- More testing is probably warranted. We have had regressions from Taffy upgrades before
- Having said that, most of the algorithm changes this cycle were driven by fixing WPT tests run through the new Servo integration. So they're possibly less likely than usual to cause regressions.

## Breaking changes

The only "breaking" change is adding a field to `Style`. Probably doesn't bear mentioning?

### CODE REVIEW

File: crates/bevy_ui/src/ui_node.rs [hunk 1]
```diff
@@ -866,6 +881,33 @@ impl Default for Display {
     }
 }
 
+/// Whether to show or hide overflowing items
```
Comment (rparrett) [inline crates/bevy_ui/src/ui_node.rs:884]: Stray comments from another `Overflow`

File: crates/bevy_ui/src/ui_node.rs [hunk 2]
```diff
@@ -271,6 +271,20 @@ pub struct Style {
     /// <https://developer.mozilla.org/en-US/docs/Web/CSS/display>
     pub display: Display,
 
+    /// Which layout algorithm to use when laying out this node's contents:
+    ///   - [`Display::Flex`]: Use the Flexbox layout algorithm
+    ///   - [`Display::Grid`]: Use the CSS Grid layout algorithm
+    ///   - [`Display::None`]: Hide this node and perform layout as if it does not exist.
+    ///
```
Comment (rparrett) [inline crates/bevy_ui/src/ui_node.rs:278]: Stray comments from `Display`

File: crates/bevy_ui/src/ui_node.rs [hunk 3]
```diff
@@ -866,6 +881,33 @@ impl Default for Display {
...
+/// Whether to show or hide overflowing items
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Reflect)]
+#[reflect(Default, PartialEq)]
+#[cfg_attr(
+    feature = "serialize",
+    derive(serde::Serialize, serde::Deserialize),
+    reflect(Serialize, Deserialize)
+)]
+
+/// Which part of a Node's box length styles like width and height control
```
Comment (rparrett) [inline crates/bevy_ui/src/ui_node.rs:893]: nit: I think we're generally placing doc comments above attributes, and not including new lines between

**Review by rparrett — CHANGES_REQUESTED** [review]

**Review by ickshonpe — APPROVED** [review]
Looks good besides the comments nits.

Adding the computed margin values is something I've wanted for a while and means I can add support for `margin-box` to #15561.

**Comment by nicoburns** [comment]
Nits addressed

**Review by rparrett — APPROVED** [review]

**Review by mnmaita — APPROVED** [review]

**Review by BenjaminBrienen — APPROVED** [review]

**Comment by robtfm** [comment]
fixes #16304

**Comment by alice-i-cecile** [comment]
Yep! I'll grab this in my next merge train.

**Pull request merged**